dashboard/test/models/contact_rollups_raw_test.rb

```diff
@@ -9,12 +9,25 @@ class ContactRollupsRawTest < ActiveSupport::TestCase
     expected_data = {opt_in: email_preference.opt_in ? 1 : 0}
     result = ContactRollupsRaw.find_by(
       email: email_preference.email,
-      sources: "dashboard.#{email_preference.class.table_name}"
+      sources: "dashboard.email_preferences"
     )
 
     assert_equal expected_data, result.data.symbolize_keys
   end
 
+  test 'extract_parent_email creates records as we would expect' do
```

Comment:
nit: grouping tests by the method-under-test will make them easier to read. You can move this test below all `extract_email_preferences` tests.